src/ast_pattern.mli

```diff
@@ -42,25 +45,26 @@ open! Import
 
     {[
       let build_payload ~loc name expr =
-        let (module B) = Ast_builder.with_loc loc in
+        let (module B) = Ast_builder.make loc in
         let open B in
-        pstr
-          [ pstr_value Nonrecursive (value_binding ~pat:(pstring name) ~expr) ]
+        [ pstr_value Nonrecursive [value_binding ~pat:(pstring name) ~expr] ]
```

Comment:
I don't know why the `Ast_builder` does not have a `pstr` function, but as is it is not building a `payload` but a `structure`.
One possibility is:
```suggestion
        Parsetree.PStr
          [ pstr_value Nonrecursive [value_binding ~pat:(pstring name) ~expr] ]
```
the other possibility is adding (back?) the `pstr` function to `Ast_builder` and using it in the example!